geode-core/src/main/java/org/apache/geode/internal/metrics/CompositeMeterRegistryFactory.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.metrics;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.composite.CompositeMeterRegistry;
+
+import org.apache.geode.annotations.VisibleForTesting;
+
+/**
+ * Creates {@code CompositeMeterRegistry} and configures commonTags.
+ */
+public interface CompositeMeterRegistryFactory {
+
+  @VisibleForTesting
+  String CLUSTER_ID_TAG = "ClusterId";
+  @VisibleForTesting
+  String MEMBER_NAME_TAG = "MemberName";
+  @VisibleForTesting
+  String HOST_NAME_TAG = "HostName";
+
+  default CompositeMeterRegistry create(int systemId, String memberName, String hostName) {
```

Comment:
Minor nitpick - this is a bit of an abuse of what default methods should be used for. No one that implements this interface can make use if your default method because they will override it if they want to do anything.

It would be better to just have a clean interface and separate DefaultCompositeMeterRegistryFactory.

Reply:
We're discovering the same thing as we add our first meters. Our next commit will almost certainly remove this default implementation and add a default concrete implementation. Is it okay with you if we do that after this PR?

Reply:
Fine by me.

Reply:
We already have this change ready for a follow-up PR that adds the 1st meter in. Should we tease that apart and port it to this PR or make the change in the next PR?

Reply:
We have one other change to make so we're going to cherry-pick this change to this PR branch.